utils/build.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Shaderc Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Builds the Shaderc project, on Linux, Darwin, or Windows.
+
+For windows, we assume we have bash and basic shell commands, and we'll
+use the MSVC compiler.
+
+For Linux and Mac, use the default C++ compiler.
+
+We require cmake and ninja commands to be present under the prebuilts
+directory in the specified root directory.
+"""
+
+import argparse
+import os
+import os.path
+import subprocess
+import sys
+
+
+def run(args, cmd, cwd):
+    """Prints a command to run, and optionally runs it.
+
+    Raises a RuntimeError if the command does not launch or otherwise fails.
+
+    Args:
+      args: Options.  If args.dry_run is true, then only print the command.
+            Otherwise run the command after printing it.
+      cmd:  List of words in the command.
+      cwd:  Working directory for the command.
+    """
+    print cmd
```

Comment:
This could be a problem if the default `python` on a user's box is python3. It's better to wrap with `()`.